src/Microsoft.Sbom.Parsers.Spdx30SbomParser/Generator.cs

```diff
@@ -615,5 +618,44 @@ private PackageVerificationCode GetPackageVerificationCode(IInternalMetadataProv
 
     public ManifestInfo RegisterManifest() => Constants.Spdx30ManifestInfo;
 
-    IDictionary<string, object> IManifestGenerator.GetMetadataDictionary(IInternalMetadataProvider internalMetadataProvider) => throw new NotSupportedException();
+    public IDictionary<string, object> GetMetadataDictionary(IInternalMetadataProvider internalMetadataProvider)
+    {
+        if (internalMetadataProvider is null)
+        {
+            throw new ArgumentNullException(nameof(internalMetadataProvider));
+        }
+
+        var generationData = internalMetadataProvider.GetGenerationData(Constants.Spdx30ManifestInfo);
+
+        var sbomToolName = internalMetadataProvider.GetMetadata(MetadataKey.SBOMToolName);
+        var sbomToolVersion = internalMetadataProvider.GetMetadata(MetadataKey.SBOMToolVersion);
+        var packageName = internalMetadataProvider.GetPackageName();
+        var packageVersion = internalMetadataProvider.GetPackageVersion();
+
+        var documentName = string.Format(Constants.SPDXDocumentNameFormatString, packageName, packageVersion);
+
+        var creationInfo = new CreationInfo
+        {
+            Created = internalMetadataProvider.GetGenerationTimestamp(),
+            CreatedBy = new List<string>
+            {
+                $"{internalMetadataProvider.GetPackageSupplier()}",
```

Comment:
Do we need to use interpolation if `GetPackageSupplier` already returns a string?